internal/cluster/kube/kube_test.go

```diff
@@ -0,0 +1,137 @@
+
+	helper_createNamespace(t, client.clientSet)
+	deploy := helper_createProxlessCompatibleDeployment(t, client.clientSet)
+
+	// error - deployment in kubernetes but not available
+	assert.Error(t, client.ScaleUpDeployment(dummyProxlessName, dummyNamespaceName, timeout))
+
+	deploy.Status.AvailableReplicas = 1
+	helper_updateDeployment(t, client.clientSet, deploy)
+
+	// no error - deployment in kubernetes and available
+	assert.NoError(t, client.ScaleUpDeployment(dummyProxlessName, dummyNamespaceName, timeout))
+}
+
+func TestClusterClient_ScaleDownDeployments(t *testing.T) {
+	client := NewCluster(fake.NewSimpleClientset())
+
+	// error - deployment is not in kubernetes
+	helper_assertAtLeastOneError(t, client.ScaleDownDeployments(dummyNamespaceName, helper_shouldScaleDown))
+
+	helper_createNamespace(t, client.clientSet)
+	deploy := helper_createProxlessCompatibleDeployment(t, client.clientSet)
+	randomDeployCreated := helper_createRandomDeployment(t, client.clientSet) // this deployment must not be scaled down
+
+	// no error - deployment in kubernetes and scaled down
+	helper_assertNoError(t, client.ScaleDownDeployments(dummyNamespaceName, helper_shouldScaleDown))
+
+	deploy.Spec.Replicas = pointer.Int32Ptr(1)
+	helper_updateDeployment(t, client.clientSet, deploy)
+
+	// no error - deployment in kubernetes and scaled up
+	helper_assertNoError(t, client.ScaleDownDeployments(dummyNamespaceName, helper_shouldScaleDown))
+
+	randomDeploy, _ := getDeployment(client.clientSet, dummyNonProxlessName, dummyNamespaceName)
+	if *randomDeploy.Spec.Replicas != *randomDeployCreated.Spec.Replicas {
+		t.Errorf("ScaleDownDeployments(); must not scale down not proxless deployment. Replicas = %d; Want = %d",
+			*randomDeploy.Spec.Replicas, *randomDeployCreated.Spec.Replicas)
+	}
+}
+
+// TODO split this test function - too much sh** here
+// the `time.sleep` are here to wait for the informer to sync
+func TestClusterClient_RunServicesEngine(t *testing.T) {
+	client := NewCluster(fake.NewSimpleClientset())
+	client.servicesInformerResyncInterval = 2
+
+	store := fakeStore{store: map[string]string{}}
+
+	helper_createNamespace(t, client.clientSet)
+	helper_createProxlessCompatibleDeployment(t, client.clientSet)
+
+	// TODO check how we wanna deal with closing the channel and stopping the routine
+	// We could use a context https://github.com/kubernetes/client-go/blob/master/examples/fake-client/main_test.go
+	// but not sure if it is worth it
+	go client.RunServicesEngine(dummyNamespaceName, store.helper_upsertStore, store.helper_deleteRouteFromStore)
+
+	// don't store random services
+	helper_createRandomService(t, client.clientSet)
+	if len(store.store) > 0 {
+		t.Errorf("RunServicesEngine(); must not store random service information")
+	}
+
+	// store proxless compatible services
+	service := helper_createProxlessCompatibleService(t, client.clientSet)
+	time.Sleep(1 * time.Second)
```

Comment:
comment - may be better to do some sort of timeout? https://stackoverflow.com/questions/52218090/is-there-a-way-to-set-test-timeout-value-dynamically/52218264

Reply:
It's not a timeout but a sleep. I need it here to leave some time for the `fake` informer to pick up the services

Reply:
I meant, I would suggest having the test have a timeout "of 5 seconds" then auto failing versus sleeping which could be error prone if the task takes a longer - and loop to see if its done.

Reply:
then I would need to split the test?
Cuz there is creation/update/delete there.

Reply:
That makes sense, not sure i'll do it in this PR but I thought about that while writing the test too.
And I could use your timeout like https://github.com/kubernetes/client-go/blob/master/examples/fake-client/main_test.go